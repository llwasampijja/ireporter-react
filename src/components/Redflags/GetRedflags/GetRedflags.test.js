import React from 'react';
import { shallow } from 'enzyme';
import { GetRedflags, mapDispatchToProps } from '.';

const mockedFunction = jest.fn();
const props = {
  redflagsList: [],
  getRedflags: mockedFunction,
};
describe('tests for the GetRedflags component', () => {
  const component = shallow(<GetRedflags {...props} />);
  it('GetRedflags renders properly', () => {
    expect(component).toMatchSnapshot();
  });

  it('loads items on mount', () => {
    const instance = component.instance();
    instance.componentDidMount();
    mapDispatchToProps(mockedFunction).getRedflags();
    expect(mockedFunction).toHaveBeenCalled();
  });
});
