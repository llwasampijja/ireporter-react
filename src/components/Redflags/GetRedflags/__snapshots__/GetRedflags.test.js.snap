// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for the GetRedflags component GetRedflags renders properly 1`] = `
<div>
  <Navbar
    adminPanel="/incidents"
    redflagsLink="Add a redflag"
    userName=""
  />
  <section
    className="section-manage-content"
    id="view-user-redflags"
  >
    <div
      className="section-hold-content"
    >
      <div>
        <div>
          No posts yet
        </div>
      </div>
    </div>
  </section>
</div>
`;
